tests/lib/rules/no-context.js

```diff
@@ -0,0 +1,155 @@
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions
+  }, {
+    code: [
+      'var Hello = function() {',
+      '  var context;',
+      '  return context;',
+      '};'
+    ].join('\n'),
+    parserOptions: parserOptions
+  }, {
+    code: [
+      'var Hello = React.createClass({',
+      '  render: function() {',
+      '    return <div>Hello {this.props.name}</div>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parser: 'babel-eslint'
+  }, {
+    code: [
+      'var Hello = function() {',
+      '  var context;',
+      '  return context;',
+      '};'
+    ].join('\n'),
+    parser: 'babel-eslint'
+  }],
+
+  invalid: [{
+    code: [
+      'var Hello = React.createClass({',
+      '  render: function() {',
+      '    return <div>Hello {this.context.name}</div>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    errors: [{
+      message: 'Using context is not allowed.'
+    }]
+  }, {
+    code: [
+      'class Hello extends React.Component {',
+      '  render() {',
+      '    return <div>Hello {this.context.name}</div>;',
+      '  }',
+      '};'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    errors: [{
+      message: 'Using context is not allowed.'
+    }]
+  }, {
+    code: [
+      'class Hello extends React.Component {',
+      '  render() {',
+      '    return <div>Hello</div>;',
+      '  }',
+      '};',
+      'Hello.contextTypes = {',
+      '  name: React.PropTypes.string',
+      '};'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    errors: [{
+      message: 'Using context is not allowed.'
+    }]
+  }, {
+    code: [
+      'class Hello extends React.Component {',
+      '  render() {',
+      '    const {name} = this.context;',
+      '    return <div>Hello {name}</div>;',
+      '  }',
+      '};'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    errors: [{
+      message: 'Using context is not allowed.'
+    }]
+  }, {
+    code: [
+      'class Hello extends React.Component {',
+      '  render() {',
+      '    const {name} = this.context;',
+      '    return <div>Hello {name}</div>;',
+      '  }',
+      '};'
+    ].join('\n'),
+    parser: 'babel-eslint',
+    errors: [{
+      message: 'Using context is not allowed.'
+    }]
+  }, {
+    code: [
+      'const Hello = (props, context) => {',
+      '  return <div>Hello {context.name}</div>;',
+      '};'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    errors: [{
+      message: 'Using context is not allowed.'
+    }]
+  }, {
+    code: [
+      'const Hello = (props, context) => {',
+      '  return <div>Hello {context.name}</div>;',
+      '};',
+      'Hello.contextTypes = {',
+      '  name: React.PropTypes.string',
+      '};'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    errors: [{
+      message: 'Using context is not allowed.'
+    }, {
+      message: 'Using context is not allowed.'
+    }]
+  }]
+});
```

Comment:
It would be nice to include examples where:

- component is an SFC that uses `function` syntax (not arrow function)
- `context` is destructured in the signature
- the second argument has a name that is not `context`
- access of context via shenanigans
  - `(...args) => { args[1] }`
  - `function Hello() { arguments[1] }`
  - `this['context']`
  - `var foo = 'context'; this[foo]`
  - `const { context } = this;`